src/ports/types.ts

```diff
@@ -0,0 +1,56 @@
+export type Undefined = null | undefined | 0 | '';
+export enum Lang {
+  /* english */
+  EN = 'en',
+  /* french */
+  FR = 'fr',
+}
+export type String = string;
+export type Number = number;
+export type ID = String | Number;
+export type Boolean = boolean;
```

Comment:
Id is a uuid therefore a string.
Also I don't think there is a reason for creating a type alias for basic types